pages/project-version/middleware/index.js

```diff
@@ -304,9 +304,23 @@ const getPreviousProtocols = (firstVersion, previousVersion, grantedVersion) =>
   const previous = get(previousVersion, 'data.protocols', []).filter(Boolean).filter(p => !p.deleted).map(p => p.id);
   const granted = get(grantedVersion, 'data.protocols', []).filter(Boolean).map(p => p.id);
   const showDeleted = uniq([ ...previous, ...granted ]);
-  const allSteps = get(previousVersion, 'data.protocols', []).filter(Boolean).filter(p => !p.deleted).map(p => p.steps);
+  function extractActiveSteps(version) {
+    return get(version, 'data.protocols', [])
+      .filter(Boolean)
+      .filter(p => !p.deleted)
+      .map(p => p.steps)
+      .map((element) => element && Array.isArray(element) ? element.filter(s => !s.deleted) : element);
+  }
+  
+  const steps = extractActiveSteps(previousVersion);
+  const firstSteps = extractActiveSteps(firstVersion);
+  const grantedSteps = extractActiveSteps(grantedVersion);
   const steps = allSteps.map((element) => element && Array.isArray(element) ? element.filter(s => !s.deleted) : element);
-  return { first, previous, granted, showDeleted, steps };
+  allSteps = get(firstVersion, 'data.protocols', []).filter(Boolean).filter(p => !p.deleted).map(p => p.steps);
+  const firstSteps = allSteps.map((element) => element && Array.isArray(element) ? element.filter(s => !s.deleted) : element);
+  allSteps = get(grantedVersion, 'data.protocols', []).filter(Boolean).filter(p => !p.deleted).map(p => p.steps);
+  const grantedSteps = allSteps.map((element) => element && Array.isArray(element) ? element.filter(s => !s.deleted) : element);
```

Comment:
This is now redundant due to change above (sorry for the confusion - the GitHub UI doesn't let you ihave suggestions that span deleted lines)

```suggestion
  allSteps = get(firstVersion, 'data.protocols', []).filter(Boolean).filter(p => !p.deleted).map(p => p.steps);
  const firstSteps = allSteps.map((element) => element && Array.isArray(element) ? element.filter(s => !s.deleted) : element);
  allSteps = get(grantedVersion, 'data.protocols', []).filter(Boolean).filter(p => !p.deleted).map(p => p.steps);
  const grantedSteps = allSteps.map((element) => element && Array.isArray(element) ? element.filter(s => !s.deleted) : element);
```

Reply:
Thanks.. I already did that..in latest commit.. Thanks for all suggestions..